File: openHAB/OpenHABWidgetMapping.m
//
//  OpenHABWidgetMapping.m
//  openHAB
//
//  Created by Victor Belov on 17/01/14.
//  Copyright (c) 2014 Victor Belov. All rights reserved.
//

#import "OpenHABWidgetMapping.h"
#import "GDataXMLNode.h"

@implementation OpenHABWidgetMapping
@synthesize command, label;

- (OpenHABWidgetMapping *) initWithXML:(GDataXMLElement *)xmlElement
{
    self = [super init];
    for (GDataXMLElement *child in [xmlElement children]) {
        if ([[self allPropertyNames] containsObject:[child name]])
            [self setValue:[child stringValue] forKey:[child name]];
    }
    return self;
}

- (OpenHABWidgetMapping *) initWithDictionary:(NSDictionary *)dictionary
{
    self = [super init];
    for (NSString *key in [dictionary allKeys]) {
        if ([[self allPropertyNames] containsObject:key])
            [self setValue:[dictionary objectForKey:key] forKey:key];
    }
    return self;
}

@end

File: openHAB/OpenHABWidgetMapping.swift
// Copyright (c) 2010-2019 Contributors to the openHAB project
//
// See the NOTICE file(s) distributed with this work for additional
// information.
//
// This program and the accompanying materials are made available under the
// terms of the Eclipse Public License 2.0 which is available at
// http://www.eclipse.org/legal/epl-2.0
//
// SPDX-License-Identifier: EPL-2.0

import Foundation
import Fuzi

class OpenHABWidgetMapping: NSObject, Decodable {
    var command = ""
    var label = ""

    init(xml xmlElement: XMLElement) {
        super.init()
        for child in xmlElement.children {
            switch child.tag {
            case "command": command = child.stringValue
            case "label": label = child.stringValue
            default:
                break
            }
        }
    }
}
